hubtools/src/lib.rs

```diff
@@ -618,13 +621,25 @@ impl RawHubrisArchive {
     pub fn write_version_to_caboose(
         &mut self,
         version: &str,
+        epoch: Option<&String>,
     ) -> Result<(), Error> {
         // Manually build the TLV-C data for the caboose
-        let data = tlvc_text::Piece::Chunk(
-            tlvc_text::Tag::new(caboose::tags::VERS),
-            vec![tlvc_text::Piece::String(version.to_owned())],
-        );
-        let out = tlvc_text::pack(&[data]);
+        let mut chunks = vec![
+            tlvc_text::Piece::Chunk(
+                tlvc_text::Tag::new(caboose::tags::VERS),
+                vec![tlvc_text::Piece::String(version.to_owned())],
+            ),
+        ];
+        if let Some(param) = epoch {
+            let epoc = param.parse::<u32>().map_err(|_| Error::InvalidEpoch(param.to_string()))?;
+            let caboose_epoc = format!("{:010}", epoc).to_owned();
+            let data = tlvc_text::Piece::Chunk(
+                tlvc_text::Tag::new(caboose::tags::EPOC),
+                vec![tlvc_text::Piece::String(caboose_epoc)],
+            );
+            chunks.push(data);
+        }
+        let out = tlvc_text::pack(&chunks);
         self.write_caboose(&out)?;
         Ok(())
     }
```

Comment:
Do we actually need this API anymore? It only seems to be used the main hubedit archive for the "no defaults" case which really feels like an API we added once and then never really cared about. I'd like to remove it unless we can make a compelling case why we want it.

Reply:
I can't find any CLI use of the --no-defaults option or direct call. It looks like it  can be removed without affecting anyone.
Done